Blahaj/Assets/Scripts/PlayerControls.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerControls : MonoBehaviour
+{
+
+    #region fields
+
+    [SerializeField] private float movementSpeed;
+    [SerializeField] private float collisionOffset = 0.05f;
+    
+    private Rigidbody2D rb2D;
+    private SpriteRenderer mySR;
+
+    private Vector2 movementInput;
+    [SerializeField] private ContactFilter2D movementFilter;
+    private List<RaycastHit2D> castCollisions = new List<RaycastHit2D>();
+
+    #endregion
+
+    private void Awake()
+    {
+        this.rb2D = GetComponent<Rigidbody2D>();
+        this.mySR = GetComponent<SpriteRenderer>();
+    }
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        this.castCollisions = new List<RaycastHit2D>();
+    }
+
+    private void OnMove(InputValue movementValue) 
+    {
+        this.movementInput = movementValue.Get<Vector2>();
+    }
+
+    private void FixedUpdate()
+    {
+        if (this.movementInput != Vector2.zero)
+        {
+            int numCollisions = rb2D.Cast(
```

Comment:
just curious so are you planning to use raycast2d to detect collisions, or how does this work?